app/controllers/application_controller.rb

```diff
@@ -1,3 +1,11 @@
 class ApplicationController < ActionController::Base
   include Authenticate
+
+  private
+
+  def ensure_authentication_allowed
```

Comment:
Consider renaming it to be specific which authentication is being checked.

For example: `ensure_session_based_authentication_allowed`